### ISSUE TITLE
Spotlight focus behaving differently depending on screen resolution.

### ISSUE THREAD

**Issue by DrakeWitt** [enactjs/enact#2993]
### Category

* [x] Rendering issue
* [ ] Crash/Exception
* [ ] Console error or warning
* [ ] CLI issue
* [ ] Other

### Description
I have multiple horizontal `VirtualList` components within a Moonstone `Scroller` component. Using keyboard navigation, the expected behavior is to change focus to the adjacent `VirtualList` (when the down key is pressed, the focus changes to the `VirtualList` below the current focused item). Whenever setting the viewport to 1920 x 1080 using Chrome's dev tools, this works. However, using my native browser resolution (2560 x 1334), different behavior occurs. Rather than changing focus to the next `VirtualList`, it attempts to scroll the page. The page has been refreshed between these tries and it makes no difference. 

Below are screen recordings:
1920 x 1080 (expected behavior):
![2022-01-10 17-09-28](https://user-images.githubusercontent.com/5975311/148847899-79fdb7b8-e0e9-4b90-8f32-9449afb203ad.gif)

Glitched behavior/ native browser resolution (2560 x 1334):
https://user-images.githubusercontent.com/5975311/148848547-3c6567ab-9704-4d61-8b3b-1248c59a3bc7.mp4

### Environment
[//]: # (Describe the relevant environment, e.g. node/npm version, browser version.)
Arch Linux (reproduced on macOS too)
Chrome Version: 97.0.4692.71 (Official Build) (64-bit)
Node: v14.17.6

**Enact version:**
v4.1.2

### Reproduction Code
(Roughly the component hierarchy)
```
<Scroller
        direction="vertical"
        verticalScrollbar="hidden"
        focusableScrollbar={false}
>
  <!-- some divs and such -->
  <VirtualList
        direction="horizontal"
        horizontalScrollbar="hidden"
        ...
  />
</Scroller>
```
### Comments and Screen Shots
See above for gifs.

**Comment by seunghoh** [comment]
Hi @DrakeWitt Thanks for reporting the issue with detailed information! We will look at it shortly. Can you share your code so that we can reproduce it?

**Comment by DrakeWitt** [comment]
I created a quick demo project that reproduces the bug: https://github.com/DrakeWitt/enact-virtuallist-bug-demo

**Comment by 0x64** [comment]
@DrakeWitt Good catch! 👍 The issue is reproduced with your sample and we're finding what actually happened inside.

**Comment by 0x64** [comment]
@DrakeWitt
Due to non-integer rectangle boundaries of focusable elements, there is some issue with choosing the next focus target on key down. We're finding a proper way to handle this and we will ship it in the next release. Until the next release, changing items smaller (at least 1px) inside lists could be helpful.
I found you don't pass `rest` params in `itemRenderer` function. It looks not an actual case but a sample code, but I left some notes for sure. If `rest` params are not passed, focus navigation inside lists does not work well. [Refer enactjs.com](https://enactjs.com/docs/developer-guide/virtual-list-scroller/#common-rules-of-items-for-virtuallistvirtualgridlist)

**Comment by 0x64** [comment]
This issue is resolved in #2997 and will be shipped in the next release.

Note that the issue occurs when a spottable element is NOT contained inside a spotlight container. Normally this does not happen since app developers make elements' size fit into a container. In the reproducing sample, elements' size is calculated with a floating-point parameter to match with viewport size and becomes slightly larger than container size. The fixed code is to allow an epsilon (under one px) on comparison between a container and an element to support various size units which lead to floating-point calculation and size mismatching.
